File: php/198-house-robber.php
<?php
class Solution {
    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function rob($nums) {
        $count = count($nums);
        if ($count == 0) {
            return 0;
        }
        $sum[0] = $nums[0];
        $sum[-1] = 0;
        $sum[-2] = 0;
        for ($i = 1; $i < $count; $i++) {
            $sum[$i] = max($sum[-1+$i], $nums[$i] + $sum[-2+$i]);
        }
        return $sum[$count-1];
    }
}

File: go/198-house-robber.go
package main

func rob(nums []int) int {
	record := make(map[int]int, len(nums))
	record[-1] = 0
	record[-2] = 0
	for i := 0; i < len(nums); i++ {
		record[i] = int(math.Max(float64(record[i-1]), float64(record[i-2]+nums[i])))
	}
	return record[len(nums)-1]
}
